src/meshapi/views/forms.py

```diff
@@ -101,6 +107,29 @@ def join_form(request: Request) -> Response:
         logging.exception("TypeError while processing JoinForm")
         return Response({"detail": "Got incomplete form request"}, status=status.HTTP_400_BAD_REQUEST)
 
+    if not DISABLE_RECAPTCHA_VALIDATION:
+        try:
+            request_source_ip, request_source_ip_is_routable = get_client_ip(request)
+            if not request_source_ip_is_routable:
+                request_source_ip = None
+
+            recaptcha_invisible_token = request.headers.get(RECAPTCHA_INVISIBLE_TOKEN_HEADER)
+            if recaptcha_invisible_token == "":
+                recaptcha_invisible_token = None
+
+            recaptcha_checkbox_token = request.headers.get(RECAPTCHA_CHECKBOX_TOKEN_HEADER)
+            if recaptcha_checkbox_token == "":
+                recaptcha_checkbox_token = None
+
+            validate_captcha_tokens(
+                recaptcha_invisible_token,
+                recaptcha_checkbox_token,
+                request_source_ip,
+            )
+        except Exception:
+            logging.exception("Captcha validation failed")
```

Comment:
Should we maybe print the exception here just in case it's not one of the "usual" ones?

Reply:
`logging.exception` does this automatically